src/server.c

```diff
@@ -3398,6 +3398,30 @@ void call(client *c, int flags) {
     server.stat_numcommands++;
 }
 
+void rejectCommand(client *c, robj *reply) {
+    flagTransaction(c);
+    if (c->cmd && c->cmd->proc == execCommand) {
+        execCommandAbort(c, reply->ptr);
+    } else {
+        /* using addReplyError* rather than addReply so that the error can be logged. */
+        addReplyErrorSds(c, reply->ptr);
+    }
+}
+
+void rejectCommandFormat(client *c, const char *fmt, ...) {
```

Comment:
It should be something like: this command is called by processCommand() every time we can't execute a command that is otherwise ready to be executed. The function will flag the transaction if any, so that EXEC fail, and report the error to the client. Or something alone these lines.